Source/WebCore/platform/SharedBuffer.cpp

```diff
@@ -121,6 +139,15 @@ Ref<SharedBuffer> FragmentedSharedBuffer::makeContiguous() const
     return SharedBuffer::create(WTFMove(combinedData));
 }
 
+#if PLATFORM(COCOA)
+std::optional<SharedMemoryHandle> FragmentedSharedBuffer::toSharedMemoryHandle() const
+{
+    RefPtr<SharedMemory> sharedMemoryBuffer = SharedMemory::copyBuffer(*this);
+    std::optional<SharedMemoryHandle> handle = sharedMemoryBuffer->createHandle(SharedMemory::Protection::ReadOnly);
```

Comment:
No need for a local variable here, just:
`return sharedMemoryBuffer->createHandle(SharedMemory::Protection::ReadOnly)`